### PR TITLE
remove node-gyp from npm

### PR DESCRIPTION
**Pull request by ajayk**
fixes #10604 and #10599  and https://github.com/wolfi-dev/os/issues/9109

### CODE REVIEW

File: npm.yaml [hunk 1]
```diff
@@ -43,14 +44,18 @@ pipeline:
...
              -name 'Makefile' -o \
              -name 'README*' -o \
              -name 'readme.markdown' \) -delete
 
       rm -rf ./*/.git* ./*/doc ./*/docs ./*/examples ./*/scripts ./*/test
-      rm -rf ./node-gyp/gyp/.git*
+      rm -rf ./node-gyp/
+      rm -rf node_modules/node-gyp
+      #rm -rf lib/node_modules/npm/node_modules/@npmcli/node-gyp/lib
+      #rm -rf lib/node_modules/npm/node_modules/@npmcli/run-script/lib/node-gyp-bin
```
Comment (pdeslaur) [inline npm.yaml:56]: Should we delete these lines altogether, or leave a comment as well?

**Review by pdeslaur — DISMISSED** [review]
One quick comment, but 🙏  thanks for fixing this!

File: npm.yaml (hunk 1 above)
Comment (ajayk) [reply npm.yaml:56]: let me delete this

**Review by pdeslaur — APPROVED** [review]
Love the new package test!

**Pull request merged**